prost-build/src/config.rs

```diff
@@ -835,37 +835,31 @@ impl Config {
         Ok(())
     }
 
-    /// Compile `.proto` files into Rust files during a Cargo build with additional code generator
-    /// configuration options.
-    ///
-    /// This method is like the `prost_build::compile_protos` function, with the added ability to
-    /// specify non-default code generation options. See that function for more information about
-    /// the arguments and generated outputs.
-    ///
-    /// The `protos` and `includes` arguments are ignored if `skip_protoc_run` is specified.
+    /// Loads `.proto` files as a `FileDescriptorSet`. This allows inspection of the descriptors
+    /// before calling `prost_build::compile_fds`. A possible use case it to compute `Config`
+    /// attributes by introspecting what is actually present in the `.proto` files before
+    /// generating the code.
     ///
     /// # Example `build.rs`
     ///
     /// ```rust,no_run
-    /// # use std::io::Result;
-    /// fn main() -> Result<()> {
-    ///   let mut prost_build = prost_build::Config::new();
-    ///   prost_build.btree_map(&["."]);
-    ///   prost_build.compile_protos(&["src/frontend.proto", "src/backend.proto"], &["src"])?;
-    ///   Ok(())
+    /// # use prost_types::FileDescriptorSet;
+    /// # use prost_build::Config;
+    /// fn main() -> std::io::Result<()> {
+    ///   let mut config = Config::new();
+    ///   let file_descriptor_set = config.load_fds(&["src/frontend.proto", "src/backend.proto"], &["src"])?;
+    ///
+    ///   // Inspect file_descriptor_set and tweak config
```

Comment:
Can you change this to a concrete example?

Reply:
Done in cbf0813.